code-review/class.User.php

```diff
@@ -0,0 +1,316 @@
+<?php
+/**
+ *
+ * Hackademic-CMS/model/common/class.User.php
+ *
+ * Hackademic User Model
+ * Class for Hackademic's User Object
+ *
+ * Copyright (c) 2012 OWASP
+ *
+ * LICENSE:
+ *
+ * This file is part of Hackademic CMS (https://www.owasp.org/index.php/OWASP_Hackademic_Challenges_Project).
+ *
+ * Hackademic CMS is free software: you can redistribute it and/or modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation, either version 2 of the License, or (at your option) any
+ * later version.
+ *
+ * Hackademic CMS is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with Hackademic CMS.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ *
+ *
+ * @author Pragya Gupta <pragya18nsit[at]gmail[dot]com>
+ * @author Konstantinos Papapanagiotou <conpap[at]gmail[dot]com>
+ * @license http://www.gnu.org/licenses/gpl.html
+ * @copyright 2012 OWASP
+ *
+ */
+require_once(HACKADEMIC_PATH."model/common/class.HackademicDB.php");
+require_once(HACKADEMIC_PATH."admin/model/class.ClassMemberships.php");
+require_once(HACKADEMIC_PATH."model/common/class.Utils.php");
+require_once(HACKADEMIC_PATH."model/common/class.ChallengeAttempts.php");
+require_once(HACKADEMIC_PATH."model/common/class.UserHasChallengeToken.php");
+
+class User {
+	public $id;
+	public $username;
+	public $password;
+	public $full_name;
+	public $type;
+	public $email;
+	public $joined;
+	public $last_visit;
+	public $is_activated;
+	public $token;
+
+	public static function findByUserName($username) {
+		global $db;
+		$sql = "SELECT * FROM users WHERE username=:username LIMIT 1";
+		$params = array(
+				':username' => $username
+			       );
+		$result_array=self::findBySQL($sql, $params);
+		return !empty($result_array)?array_shift($result_array):false;
+	}
+
+	public static function getUser($id) {
+		global $db;
+		$sql = "SELECT * FROM users WHERE id = :id LIMIT 1";
+		$params = array(
+				':id' => $id
+			       );
+		$result_array=self::findBySQL($sql, $params);
+		return !empty($result_array)?array_shift($result_array):false;
```

Comment:
### Pažeidžiamos taisyklės
- [ ] Neturėtų būti naudojami if sakiniai su neiginiu
- [ ] Kodas turėtų vadovautis lintinimo standartais
### Komentarai
Reikėtų perrašyti sąlygą be neiginio, `?` ir `:` išskirti tarpais:
```suggestion
		return empty($result_array)? false : array_shift($result_array);
```